src/platform/packages/shared/presentation/presentation_containers/interfaces/panel_management.ts

```diff
@@ -40,3 +42,14 @@ export const apiCanPinPanel = (api: unknown): api is CanPinPanel => {
     typeof (api as CanPinPanel)?.unpinPanel === 'function'
   );
 };
+
+export type CanAddPinnedPanel = CanAddNewPanel &
```

Comment:
Why did you choose to define this separately rather than add it to the `apiCanPinPanel` interface?

Reply:
I don't know if there's a scenario when an api can pin a panel but not add a new panel, but since it's theoretically possible I figured we should guard against it

Reply:
I disagree. If an API has pinnable panels, then we should expect that there is a way to add a pinned panel directly. And since this is only used by Dashboard and there are no plans to expand this at this point, I would prefer we keep new interfaces to a minimum rather than planning for something that is unlikely to happen in the near future. What do you think?